File: Capstone/Backend/AssessmentPlatform/src/main/java/com/nucleusteq/asessmentPlatform/entities/Result.java
package com.nucleusteq.asessmentPlatform.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Entity class representing a result of a quiz attempt.
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Result {
    /**
     * The unique identifier of the result.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "result_id")
    private int resultId;

    /**
     * The number of marks obtained in the quiz.
     */
    private int obtainedMarks;

    /**
     * The total number of marks available in the quiz.
     */
    private int totalMarks;

    /**
     * The username of the user who attempted the quiz.
     */
    private String userName;

    /**
     * The email address of the user who attempted the quiz.
     */
    private String email;

    /**
     * The title of the quiz.
     */
    private String quizTitle;

    /**
     * The title of the quiz category.
     */
    private String categoryTitle;

    /**
     * The number of questions attempted in the quiz.
     */
    private int numofAttemptedQues;

    /**
     * The total number of questions in the quiz.
     */
    private int totalNumOfQues;

    /**
     * The date and time when the quiz was attempted.
     */
    private String dateAndTime;
}

File: Capstone/Frontend/assessment-platform-frontend/src/Component/Result/Result.js
import React, { useState, useEffect } from 'react'
import Navbar from '../Navbar/Navbar';
import DisableBackButton from '../DisableBackButton';
import ServiceURL from '../Service/ServiceURL';
export default function Result() {
    const[results, setResults]= useState([]);
    const role = localStorage.getItem('userRole');
    const userEmail = localStorage.getItem('selectedEmail');

    useEffect( () => {
      if(role === 'user')
        loadResults();
      if(role === 'admin')
      loadAllResult();
    },[role]);
    
    const loadResults = async () => {
        try {
          const response = await ServiceURL.getResultByUserEmail(userEmail);
          setResults(response.data);
        } catch (error) {
          console.error("Error loading results:", error);
        }
      };
    const loadAllResult = async () =>{
      try{
        const response = await ServiceURL.getResults();
        setResults(response.data);
      }catch(error){
        console.error("Error loading results:", error);
      }
    }
  return (
  <>
    <Navbar/>
    <DisableBackButton/>
          <div className="category-wrapper-container">
            <div className="category-card">
              <div className="category_wrapper">
              <h2 className="cat_head">Result</h2>
              {role === 'user' &&   <h2 className="cat_head"> Welcome  {userEmail}</h2>}
                <table className="category-table">
                  <thead>
                    <tr>
                    <th>ID</th>
                    {role === 'admin' && <>
                    <th>Name</th>
                    <th>Email</th></>}
                    <th>Category</th>
                    <th>Quiz</th>
                    <th>Obtained Marks</th>
                    <th>Total Marks</th>
                    <th>Number of Question Attempted</th>
                    <th>Total Question</th>
                    <th><center>Date & Time</center></th>
                    </tr>
                  </thead>
                  <tbody>
                    {results.map((result, index) => (
                      <tr key={index}>
                        <td>{++index}</td>
                        {role === 'admin' && <>
                        <td>{result.userName}</td>
                        <td>{result.email}</td></>}
                        <td>{result.categoryTitle}</td>
                        <td>{result.quizTitle}</td>
                        <td><center>{result.obtainedMarks}</center></td>
                        <td><center>{result.totalMarks}</center></td>
                        <td><center>{result.numofAttemptedQues}</center></td>
                        <td><center>{result.totalNumOfQues}</center></td>
                        <td>{result.dateAndTime}</td>
                      </tr>
                    ))}
                  </tbody>
                </table>
              </div>
            </div>
          </div></>
  )
}
